src/pipeline/utils.py

```diff
@@ -14,4 +18,18 @@ def get_args(yaml_file_path):
             (ModelArguments, DataArguments, SFTConfig)
         ).parse_yaml_file(yaml_file_path)
     
-    return configs
\ No newline at end of file
+    return configs
+
+def push_to_hf_hub(dataset_id, ds, hf_token, append=True):
+    exist = False
+
+    try:
+        create_repo(dataset_id, repo_type="dataset", token=hf_token)
```

Comment:
In case a repo was mistakenly created, we can do "exist_ok=True" to not raise an error.

Reply:
good point. didn't know about that flag

Reply:
if dataset is empty, `load_dataset()` fails with `EmptyDatasetError` exception. In this case, the dataset should be directly pushed to the target repository without pre-loading the existing ones from the hub with `load_dataset()`.

I can not find a way to know if the dataset from the remote repository is empty or not when `exist_ok=True`. 
- Without `exist_ok`, the `create_repo()` throws an error when the repository was already created. In this case, I can determine whether or not to use `load_dataset()`. 

If you know a better way to handle this, please let me know. If no, I think current implementation should be kept.

Reply:
Okay then I think we should add a utility to delete a dataset repo that was maybe created mistakenly. We can catch it as an error in the except block and call [`delete_repo()`](https://huggingface.co/docs/huggingface_hub/v0.22.2/en/package_reference/hf_api#huggingface_hub.HfApi.delete_repo) there. This should work better than deleting a faulty repo manually. WDYT?

Reply:
Then this should be a separate CLI command I guess?

Reply:
Why? 

```python
try:
    # Attempt to create a repo with `exist_ok=True`
    ...
# get the exact error name from the docs
except RepoAlreadyExistsError:
    delete_repo(...)
```

Should be this simple and doesn't have to be complicated if I am not missing out on something.

Reply:
Oh this function is able to append to the existing dataset when there is already existing repo. Since batch inference results could be accumulated.

But your suggestion is to delete the repo whenever there is existing one, do I understand correctly?

Reply:
Oh I see. But `create_repo()` doesn't throw an error if a repo already exists (without `exist_ok=True`?

How are we accumulating the results of batch inference? This is the result of running the fine-tuned LLM on an eval split, right? If that's the case there are multiple options:

* Keep gathering the results until we're done and then create a dataset, and push it. 
* As soon as we gather results on a batch, create a dataset, and push it. 

For the second option, we can do:

```python
try:
   # load the existing dataset
   dataset = load_dataset(dataset_id, split=...)
   
   # create temporary dataset on the incoming results.
   temp_dataset = ...
   final_dataset = concatenate_datasets([dataset, temp_dataset])
   final_dataset.push_to_hub(...)
   
except RepoNoutFoundError:
   create_repo(...)
   create_dataset(...)
   push_to_hub(...)
```

Does this make sense? 

`concatenate_datasets`: https://huggingface.co/docs/datasets/en/process#concatenate

Reply:
yup.

I think your code and mine basically do the same thing (2nd option).
maybe I should add docstring on it (that is why we had some misunderstanding I guess).

Reply:
just added some docstring